app/controllers/green_lanes/applicable_exemptions_controller.rb

```diff
@@ -38,22 +33,12 @@ def determine_category
 
     # Goods nomenclature methods
     def goods_nomenclature
-      @goods_nomenclature ||= GreenLanes::GoodsNomenclature.find(
-        params[:commodity_code],
-        {
-          filter: {
-            geographical_area_id: params[:country_of_origin],
-            moving_date: params[:moving_date],
-          },
-          as_of: params[:moving_date],
-        },
-        { authorization: TradeTariffFrontend.green_lanes_api_token },
-      )
+      @goods_nomenclature ||= FetchGoodsNomenclature.new(goods_nomenclature_params).call
     end
 
     # Form handling methods
-    def exemptions_form
-      ApplicableExemptionsForm.new(exemptions_params)
+    def set_exemptions_form
+      @exemptions_form = ApplicableExemptionsForm.new(exemptions_params)
```

Comment:
keeping the assignment in the action  makes it clear what are the 
instance vars that are used in the views.

This way (before filter, calling a method assigning the instance var) is a bit obscure.

Reply:
il revert back